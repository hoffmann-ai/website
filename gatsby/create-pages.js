'use strict';

const path = require('path');
const _ = require('lodash');
const createCategoriesPages = require('./pagination/create-categories-pages.js');
const createTagsPages = require('./pagination/create-tags-pages.js');
const createPostsPages = require('./pagination/create-posts-pages.js');

const createPages = async ({ graphql, actions }) => {
  const { createPage } = actions;

  // 404
  createPage({
    path: '/404',
    component: path.resolve('./src/templates/not-found-template.js'),
  });

  // Tags list
  createPage({
    path: '/tags',
    component: path.resolve('./src/templates/tags-list-template.js'),
  });

  // Categories list
  createPage({
    path: '/categories',
    component: path.resolve('./src/templates/categories-list-template.js'),
  });

  const lastPosts = await graphql(`
    {
      allMarkdownRemark(
        filter: { frontmatter: { template: { eq: "post" } } }
        limit: 4
        sort: { fields: frontmatter___date, order: DESC }
      ) {
        edges {
          node {
            fields {
              slug
            }
            frontmatter {
              date
              description
              title
              category
              socialImage {
                childImageSharp {
                  gatsbyImageData(width: 450, height: 164)
                }
              }
            }
          }
        }
      }
    }
  `);

  const skills = await graphql(`
    {
      allMarkdownRemark(filter: {frontmatter: {template: {eq: "skills"}}}) {
        edges {
          node {
            frontmatter {
              name
              description
              logo {
                childImageSharp {
                  gatsbyImageData(width: 48, height: 48)
                }
              }
            }
          }
        }
      }
    }
  `);

  const clients = await graphql(`
    {
      allFile(filter: { relativeDirectory: { eq: "clients" } }) {
        edges {
          node {
            childImageSharp {
              gatsbyImageData(height: 48)
            }
            name
          }
        }
      }
    }
  `);

  const landing = await graphql(`
    {
      allFile(filter: { relativeDirectory: { eq: "landing" } }) {
        edges {
          node {
            childImageSharp {
              gatsbyImageData(height: 300)
            }
            name
          }
        }
      }
    }
  `);

  const services = await graphql(`
  {
    allMarkdownRemark(filter: {frontmatter: {template: {eq: "services"}}}) {
      edges {
        node {
          frontmatter {
            name
            description
            logo {
              childImageSharp {
                gatsbyImageData(height: 500)
              }
            }
          }
        }
      }
    }
  }
`);

  // Home page
  createPage({
    path: '/',
    component: path.resolve('./src/templates/landing-template.js'),
    context: {
      lastPosts, skills, clients, landing, services
    },
  });

  // Posts and pages from markdown
  const result = await graphql(`
    {
      allMarkdownRemark(filter: { frontmatter: { draft: { ne: true } } }) {
        edges {
          node {
            frontmatter {
              template
            }
            fields {
              slug
            }
          }
        }
      }
    }
  `);

  const { edges } = result.data.allMarkdownRemark;

  _.each(edges, (edge) => {
    if (_.get(edge, 'node.frontmatter.template') === 'page') {
      createPage({
        path: edge.node.fields.slug,
        component: path.resolve('./src/templates/page-template.js'),
        context: { slug: edge.node.fields.slug },
      });
    } else if (_.get(edge, 'node.frontmatter.template') === 'post') {
      createPage({
        path: edge.node.fields.slug,
        component: path.resolve('./src/templates/post-template.js'),
        context: { slug: edge.node.fields.slug },
      });
    }
  });

  // Feeds
  await createTagsPages(graphql, actions);
  await createCategoriesPages(graphql, actions);
  await createPostsPages(graphql, actions);
};

module.exports = createPages;
