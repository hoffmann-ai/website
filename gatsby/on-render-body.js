'use strict';

const React = require('react');

const onRenderBody = ({ setPostBodyComponents }) => {
  setPostBodyComponents([
    <script
      key="axeptio"
      dangerouslySetInnerHTML={{
        __html: `
          window.axeptioSettings = {
            clientId: "60f291c2afa9d02a7f02768e",
            cookiesVersion: "hoffmann.ai-base",
          };
          (function(d, s) {
            var t = d.getElementsByTagName(s)[0], e = d.createElement(s);
            e.async = true;
            e.src = "//static.axept.io/sdk-slim.js";
            t.parentNode.insertBefore(e, t);
          })(document, "script");
      `,
      }}
    />
  ]);
};

module.exports = onRenderBody;
