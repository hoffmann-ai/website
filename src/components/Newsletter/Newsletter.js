// @flow strict
import React, { useState } from 'react';
import addToMailchimp from 'gatsby-plugin-mailchimp';
import styles from './Newsletter.module.scss';

type Props = {
  style: string,
};

const Newsletter = ({ style }: Props) => {
  const [validationMessage, setValidationMessage] = useState('');
  const [email, setEmail] = useState('');
  const validate = () => {
    setValidationMessage('Merci pour votre abonnement.');
    setEmail('');
  };
  const changeEmailHandler = (event) => {
    setEmail(event.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    addToMailchimp(email)
      .then(() => {
        validate();
      })
      .catch(() => {
        setValidationMessage('Une erreur est survenue.');
      });
  };

  return (
    <div className={styles['newsletter__module']}>
      <div className={styles['newsletter__module__inner']}>
        <div className={styles['newsletter__module__form']}>
          <form
            className={
              style === 'footer'
                ? styles['newsletter__module__form__subscribe__footer']
                : styles['newsletter__module__form__subscribe']
            }
            onSubmit={handleSubmit}
          >
            <input
              value={email}
              className={styles['newsletter__module__form__subscribe__input']}
              type='email'
              onChange={changeEmailHandler}
              name='email'
              id='email'
              placeholder='Email'
              required
            />
            <button
              className={styles['newsletter__module__form__subscribe__submit']}
              type='submit'
            >
              S'abonner
            </button>
            <h6 className={styles['newsletter__module__form__message']}>
              {validationMessage}
            </h6>
          </form>
        </div>
      </div>
    </div>
  );
};

export default Newsletter;
