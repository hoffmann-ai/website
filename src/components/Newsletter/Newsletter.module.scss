@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';
@import '../../assets/scss/base/generic';
* {
  --primary-color: #0873e8;
  --text-color-dark-light: #d4d4d4;
  --text-color-dark: #4a4a4a;
  // padding:0;
  // margin:0;
}
.newsletter__module {
  max-width: 100%;
  width: 100%;
  color: #222;
  line-height: 1.625;
  font-size: 1rem;
  background-color: transparent;
  padding: 0% 0px 0% 0px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.newsletter__module__inner{
  max-width: 100%;
  max-height: 120px;
}
.newsletter__module__form__subscribe, .newsletter__module__form__subscribe__navbar {
  max-width: 100%;
  background-color: transparent;
  width: 100%;
  display: grid;
  margin: auto;
  grid-template-columns: 60% 50%;
}
.newsletter__module__form__subscribe__navbar {
  width: 80%;
}
.newsletter__module__title{
  margin-top: 0px;
}
.newsletter__module__form__subscribe__input {
  font-size: 1rem;
  height: 35px;
  border-radius: 3px;
  border: 1px solid rgb(146, 146, 146, 0.5);
  padding-left: 8px;
  background-color: #FFF;
}
.newsletter__module__form__subscribe__submit {
  height: 39px;
  margin-left: 7%;
}
@media screen and (max-width: 900px) {
  .newsletter__module__form__subscribe__navbar {
    display:none;
  }
  .newsletter__module__form__subscribe{
    width:76%;
  }
  .newsletter__module__form__subscribe__submit {
    margin-left: 0px;
  }
}   
.newsletter__module__form__message{
  margin:0;
}
