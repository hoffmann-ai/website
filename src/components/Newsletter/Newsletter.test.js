// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Newsletter from './Newsletter';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../types';

describe('Newsletter', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => render(siteMetadata),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Newsletter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
