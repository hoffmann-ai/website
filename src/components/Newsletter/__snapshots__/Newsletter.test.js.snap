// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter renders correctly 1`] = `
<div
  className="newsletter__module"
>
  <div
    className="newsletter__module__inner"
  >
<<<<<<< HEAD
    <h4>
      Abonnez-vous à notre newsletter !
    </h4>
=======
    <h5>
      Abonnez-vous à notre newsletter !
    </h5>
>>>>>>> fa05346 (feat article on mailchimp usage)
    <div
      className="newsletter__module__form"
    >
      <form
        className="newsletter__module__form__subscribe"
        onSubmit={[Function]}
      >
        <input
          className="newsletter__module__form__subscribe__input"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="votre email..."
          required={true}
          type="email"
          value=""
        />
        <button
          className="newsletter__module__form__subscribe__submit"
          type="submit"
        >
          S'abonner
        </button>
        <h6>
          
        </h6>
      </form>
    </div>
  </div>
</div>
`;
